prereise/gather/demanddata/uci/HiGrid_Elec_Trans4_8_nths.py

```diff
@@ -0,0 +1,195 @@
+        k = month_first_day[i]
+
+        # Iterate over the day of each month
+        for j in range(month_days[i]):
+
+            # if k < 6: weekday, if k >= 6: weekend
+            if k < 6:
+                inputDay[D_count] = 2
+            elif k >= 6:
+                inputDay[D_count] = 1
+
+            # restarts the week counter independent of the intial day
+            if k == 7:
+                k = 1
+            else:
+                k += 1
+
+            # increment day counter
+            D_count += 1
+
+    return inputDay
+
+
+def get_inputMonth(month_days: list[int], month_first_day: list[int]) -> np.ndarray:
+    '''determines month of each day
+
+    :param list[int] month_days: a list of integers where each value in the list represents the number of days in that month (0 indexed)
+    :param list[int] month_first_day: a list of integers where each value in the list represents the day of the week of the first day of that month (0 indexed)
+    :return: np.ndarray -- stores the month the day is in for each day of the year
+
+    '''
+    # initalize day counter
+    D_count = 0
+
+    # initialize integers because all zeros
+    inputMonth = np.zeros(365, int)
+
+    # iterate over 12 months
+    for i in range(len(month_days)):
+
+        #initialize first day of each month for use in the next for loop
+        k = month_first_day[i]
+
+        #Iterate over the day of each month
+        for j in range(month_days[i]):
+            # Populate inputMonth with month of each day ex: inputMonth[100] gives us month number of day 101
+            inputMonth[D_count] = i
+
+            #increment day counter
+            D_count += 1
+
+    return inputMonth
+
+
+def get_month2(data: np.array(list[float or int]])) -> np.array(int):
+    '''gets month2 value from data
+    :param np.array(list[float or int]) data: the data to get months from
+    :return: np.array(int) -- list of months
+    '''
+    return np.array([x[5] % 100 for x in data], int)
+
+
+def get_dayofweek2(data: np.array(list[float or int]])) -> np.array(int):
+    '''gets dayofweek2 value from data
+    :param np.array(list[float or int]) data: the data to get day of week from
+    :return: np.array(int) -- the day of the week for each entry in data
+    '''
+    return np.array([x[6] for x in data], int)
+
+def get_day2(data: np.array(list[float or int]])) -> np.array(int):
+    '''gets day2 value from data
+    :param np.array(list[float or int]) data: the data to get day of week from
+    :return: np.array(int) -- indicates weekend or weekday for every day
+    '''
+    return np.array([x[7] for x in data])
+
+def load_data(census_1: int) -> list[list[float or int]]:
+    '''loads the data at nhts_census.mat
+    :param int census_1: the type of census
+    :return: list[list] -- the data loaded from nths_census.mat
+    '''
+    # somehow load in 'nhts_census.mat'
+    nhts_census = scipy.io.loadmat('nhts_census.mat')

```

Comment:
It would be better to pass the path as an explicit argument to this function, otherwise we can only execute this from a certain location and we have limits on the name of the file.

Reply:
`scipy` is not imported